iai-callgrind-runner/src/runner/tool/logfile_parser.rs

```diff
@@ -44,23 +46,97 @@ pub struct LogfileParser {
 #[derive(Debug, Clone)]
 pub struct LogfileSummary {
     pub command: PathBuf,
+    pub old_pid: Option<i32>,
+    pub old_parent_pid: Option<i32>,
     pub pid: i32,
     pub parent_pid: Option<i32>,
     pub fields: Vec<(String, String)>,
     pub details: Vec<String>,
     pub error_summary: Option<ErrorSummary>,
+    pub cost_summary: Option<CostsSummary<String>>,
     pub log_path: PathBuf,
 }
 
+#[derive(Default, Debug, Clone)]
+pub struct OldLogfileSummary {
+    pub old_pid: Option<i32>,
+    pub old_parent_pid: Option<i32>,
+    pub old_costs: Option<Costs<String>>,
+}
+
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
 enum State {
     Header,
     HeaderSpace,
     Body,
 }
 
-impl LogfileParser {
-    fn parse_single(&self, path: PathBuf) -> Result<LogfileSummary> {
+pub trait LogfileParserT {
```

Comment:
I'm not convinced this trait and all the different parse methods are necessary. What was wrong with the old `parse_single` here and `impl Parser for LogfileParser` later returning (multiple) `LogfileSummary`? As already mentioned above, the `CostsSummary` can be calculated later from the `Costs` contained in the `LogfileSummary`. The `CostsSummary` can then be stored in the `ToolRunSummary`. For now, you can only store the new `pid`, `parent_pid`, `details` etc. in the `ToolRunSummary`.

Reply:
I changed the `LogfileT` trait to make it more similar to `Parser`, but kept it as a separate trait so that it could have a required method for parsing a single file, and have a default implementation to handle multiple files.

Reply:
Is there anything else you want me to change at the same time?